kotlin-sample/app/src/main/java/com/here/android/example/kotlin/AppMapView.kt

```diff
@@ -0,0 +1,303 @@
+                                    )
+                                } ?: kotlin.run {
+                                    Toast.makeText(
+                                        this@AppMapView,
+                                        "Error:route results returned is not valid",
+                                        Toast.LENGTH_LONG
+                                    ).show()
+                                }
+                            }
+                        } else {
+                            Toast.makeText(
+                                this@AppMapView,
+                                "Error: route calculation returned error code: $routingError",
+                                Toast.LENGTH_LONG
+                            ).show()
+                        }
+                    }
+                })
+            return true
+        } else {
+            Toast.makeText(this, "Please choose minimum 2 waypoints", Toast.LENGTH_LONG)
+                .show()
+            return false
+        }
+    }
+
+    private fun startSimulation() {
+        map.positionIndicator.isVisible = true
+
+        navigationManager =
+            if (mapViewModel.getNavigationManager() != null) {
+                mapViewModel.getNavigationManager().also {navigationManager ->
+                    navigationManager!!.setMap(map)
+                    navigationManager.resume()
+                }
+            }
+            else {
+                NavigationManager.getInstance().also {navigationManager ->
+                    mapViewModel.setNavigationManager(navigationManager)
+                    navigationManager.setMap(map)
+                    navigationManager.mapUpdateMode = NavigationManager.MapUpdateMode.ROADVIEW_NOZOOM
+                    route?.let { navigationManager.simulate(it, 40) }
+                }
+            }
+
+        map.tilt = 30f
+        map.zoomLevel = map.maxZoomLevel / 1.1
+    }
+
+    fun onClick(title: String): String {
+        if (buttonTitles.text != title) {
+            buttonTitles = when(title) {
+                ButtonTitles.CREATE_ROUTE.text -> ButtonTitles.CREATE_ROUTE
+                ButtonTitles.SIMULATE.text -> ButtonTitles.SIMULATE
+                ButtonTitles.FINISH_SIMULATION.text -> ButtonTitles.FINISH_SIMULATION
+                else -> ButtonTitles.CREATE_ROUTE
+            }
+        }
+        buttonTitles = when (buttonTitles) {
+            ButtonTitles.CREATE_ROUTE -> {
+                if (createRoute()) ButtonTitles.SIMULATE else ButtonTitles.CREATE_ROUTE
+            }
+            ButtonTitles.SIMULATE -> {
+                startSimulation()
+                ButtonTitles.FINISH_SIMULATION
+            }
+            ButtonTitles.FINISH_SIMULATION -> {
+                clearMapData()
+                ButtonTitles.CREATE_ROUTE
+            }
+        }
+
+        isRouteCalculated = buttonTitles == ButtonTitles.CREATE_ROUTE
```

Comment:
Integer ids( e.g. R.id.buttion_id) should be compared, not the title of the button.

Reply:
But I have one button with few titles and this button don't need to have an id